src/libwalletqt/WalletManager.cpp

```diff
@@ -86,6 +86,21 @@ Wallet *WalletManager::recoveryWallet(const QString &path, const QString &memo,
     return m_currentWallet;
 }
 
+Wallet *WalletManager::createWalletFromKeys(const QString &path, const QString &language, bool testnet,
+                                            const QString &address, const QString &viewkey, const QString &spendkey,
+                                            quint64 restoreHeight)
+{
+    QMutexLocker locker(&m_mutex);
+    if (m_currentWallet) {
+        qDebug() << "Closing open m_currentWallet" << m_currentWallet;
+        delete m_currentWallet;
```

Comment:
Not sure if i understand what you mean. If wallet is created with error it will be deleted by caller. https://github.com/monero-project/monero-core/pull/437/files/83f617e4861dbb6abdbeb678db8d14f9ef963dfa#diff-15b9367dcbcc645d6ebb7fd91e46e6e1R98